plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/loading/PluginInstancesLoader.java

```diff
@@ -169,25 +192,30 @@ Map<String, Plugin> instantiatePluginClasses(Map<PluginClassLoaderDef, ClassLoad
     return instancesByPluginKey;
   }
 
-  public void unload() {
-    for (ClassLoader classLoader : classloadersToClose) {
-      if (classLoader instanceof Closeable) {
-        try {
-          ((Closeable) classLoader).close();
-        } catch (Exception e) {
-          LOG.error("Fail to close classloader", e);
+  @Override
+  public void close() throws IOException {
+    Queue<IOException> exceptions = new LinkedList<>();
+    synchronized (classloadersToClose) {
+      for (var classLoader : classloadersToClose) {
+        if (classLoader instanceof Closeable) {
+          tryAndCollectIOException(((Closeable) classLoader)::close, exceptions);
         }
       }
+      classloadersToClose.clear();
+    }
+    synchronized (jarFilesToClose) {
+      for (var jarFile : jarFilesToClose) {
+        tryAndCollectIOException(jarFile::close, exceptions);
+      }
+      jarFilesToClose.clear();
```

Comment:
If this is the workaround because of the JDK, a small comment somewhere might be useful with the link to the ticket